content/en/docs/configuration/selfsigned.md

```diff
@@ -36,25 +40,114 @@ metadata:
   namespace: sandbox
 spec:
   selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: selfsigned-cluster-issuer
+spec:
+  selfSigned: {}
 ```
-Optionally, you can specify [CRL](https://en.wikipedia.org/wiki/Certificate_revocation_list) Distribution Points. An array of strings each of which identifies the location of the CRL from which the revocation of this certificate can be checked:
+
+Once deployed, you should be able to see immediately that the issuer is ready
+for signing:
+
+```bash
+$ kubectl get issuers  -n sandbox -o wide selfsigned-issuer
+NAME                READY   STATUS                AGE
+selfsigned-issuer   True                          2m
+
+$ kubectl get clusterissuers -o wide selfsigned-cluster-issuer
+NAME                        READY   STATUS   AGE
+selfsigned-cluster-issuer   True             3m
+```
+
+### Bootstrapping `CA` Issuers
+
+One of the ideal use cases for `SelfSigned` issuers is to bootstrap a custom
+root certificate for a private PKI, including with the cert-manager [`CA`](../ca/)
+issuer.
+
+The below YAML will create a `SelfSigned` issuer, issue a root certificate and
+in use that root as a `CA` issuer:
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: selfsigned-issuer
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: osm-ca
+  namespace: sandbox
+spec:
+  isCA: true
+  commonName: osm-system
+  secretName: osm-ca
+  issuerRef:
+    name: selfsigned-issuer
+    kind: ClusterIssuer
+    group: cert-manager.io
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: osm-ca
+  namespace: sandbox
+spec:
+  ca:
+    secretName: osm-ca
 ```
+
+### CRL Distribution Points
+
+You may also optionally specify [CRL](https://en.wikipedia.org/wiki/Certificate_revocation_list)
+Distribution Points as an array of strings, each of which identifies the location of a CRL in
+which the revocation status of issued certificates can be checked:
```

Comment:
```suggestion
You may also optionally specify [CRL](https://en.wikipedia.org/wiki/Certificate_revocation_list)
Distribution Points as an array of URLs, each of which identifies the location of a CRL in
which the revocation status of issued certificates can be checked:
```

Do they have to be URLs? I assumed so.

Reply:
You raise a good point that I'd not considered: in the spec, they're `DistributionPoint`s: https://tools.ietf.org/html/rfc5280#section-4.2.1.13.

We don't support the full array of options there (and I imagine most implementations of TLS probably don't either). In practice, I can't imagine anyone wanting to specify something here that's not a URL (or URI), so my change from "strings" to "URLs" is probably correct and helpful.

(Honestly, I think my preference would be to remove any reference in the docs to cRLDistributionPoints for SelfSigned issuers because I can't imagine any situation where it would make sense to include it...)

Reply:
Ok. So we'll stick with strings then.